src/runtime/runtime.go

```diff
@@ -88,11 +88,8 @@ func LockOSThread() {
 func UnlockOSThread() {
 }
 
+//go:noinline
 func KeepAlive(x interface{}) {
-	// Unimplemented. Only required with SetFinalizer().
-}
-
-func SetFinalizer(obj interface{}, finalizer interface{}) {
 	// Unimplemented.
 }
```

Comment:
`//go:noinline` doesn't do anything here, except avoid calling a no-op function. Most importantly, this does nothing to keep an object alive: LLVM will happily remove the `x` parameter because it is not used.

I think the only reliable way to implement this is either by using a volatile operation, or by calling an assembly stub (that simply returns).

Reply:
Thanks for the hint, I went ahead and removed the inline and added a TODO based on it. Would like to dig into it in a followup